node-hub/opencv-video-capture/main.py

```diff
@@ -0,0 +1,96 @@
+import os
+import argparse
+import cv2
+
+import numpy as np
+import pyarrow as pa
+
+from dora import Node
+
+
+def main():
+    # Handle dynamic nodes, ask for the name of the node in the dataflow, and the same values as the ENV variables.
+    parser = argparse.ArgumentParser(
+        description="OpenCV Video Capture: This node is used to capture video from a camera.")
+
+    parser.add_argument("--name", type=str, required=False, help="The name of the node in the dataflow.",
+                        default="opencv-video-capture")
+    parser.add_argument("--path", type=int, required=False,
+                        help="The path of the device to capture (e.g. /dev/video1, or an index like 0, 1...", default=0)
+    parser.add_argument("--image-width", type=int, required=False,
+                        help="The width of the image output. Default is the camera width.", default=None)
+    parser.add_argument("--image-height", type=int, required=False,
+                        help="The height of the camera. Default is the camera height.", default=None)
+
+    args = parser.parse_args()
+
+    video_capture_path = os.getenv("CAPTURE_PATH", args.path)
+
+    if isinstance(video_capture_path, str) and video_capture_path.isnumeric():
+        video_capture_path = int(video_capture_path)
```

Comment:
So opencv video capture path is not necessarely a string. It can be an IP address and a link.

Could be good to leave those possibilities open.

Reply:
So with the current implementation you can pass "/dev/video0", or "COM6" or "0", only "0" will be convert to an int

Reply:
Ah! I missread it! you're right!